src/vs/workbench/parts/debug/browser/exceptionWidget.ts

```diff
@@ -33,15 +33,40 @@ export class ExceptionWidget extends ZoneWidget {
 		const fontInfo = this.editor.getConfiguration().fontInfo;
 		this.container.style.fontSize = `${fontInfo.fontSize}px`;
 		this.container.style.lineHeight = `${fontInfo.lineHeight}px`;
-
-		let title = $('.title');
-		title.textContent = nls.localize('exceptionThrown', 'Exception occurred');
-		dom.append(container, title);
-
 		const thread = this.debugService.getViewModel().focusedThread;
+
 		if (thread && thread.stoppedDetails) {
+			let title = $('.title');
 			let msg = $('.message');
-			msg.textContent = thread.stoppedDetails.text;
+
+			if (this.exceptionInfo) {
+				let conditionMessage;
+				switch (this.exceptionInfo.body.breakMode) {
+					case 'never':
+						conditionMessage = nls.localize('neverException', 'User-handled exception has occurred.');
+						break;
+					case 'always':
+						conditionMessage = nls.localize('alwaysException', 'Always-breaking exception has occurred.');
+						break;
+					case 'unhandled':
+						conditionMessage = nls.localize('unhandledException', 'Unhandled exception has occurred.');
+						break;
+					case 'userUnhandled':
+						conditionMessage = nls.localize('userUnhandledException', 'User-unhandled exception has occurred.');
+						break;
+					default:
+						conditionMessage = '';
+						break;
+				}
+
+				title.textContent = `${conditionMessage} ${this.exceptionInfo.body.description}`;
```

Comment:
I did not put it explicitly and used a dot on the conditionMessage instead, because description itself contains a colon after the exception type, so that it ends up as `Unhandled exception has occurred: Error: test message` which looks ugly with two colons in my opinion. 
So currently it is printed like that: 
`Unhandled exception has occurred. Error: test message`
Although I can use quotes on the description to make it look better with two colons:  
`Unhandled exception has occurred: 'Error: test message'`

What do you think is better?